src/main/kotlin/app/revanced/patches/youtube/layout/theme/patch/ThemePatch.kt

```diff
@@ -25,30 +23,105 @@ class ThemePatch : ResourcePatch {
-                    "material_grey_white_1000", "sud_glif_v3_dialog_background_color_dark" -> darkThemeBackgroundColor
+                    "yt_black4", "yt_status_bar_background_dark", "material_grey_850" -> darkThemeBackgroundColor
 
-                    "yt_white1", "yt_white1_opacity95", "yt_white1_opacity98", "yt_white2", "yt_white3", "yt_white4",
-                    "sud_glif_v3_dialog_background_color_light" -> lightThemeBackgroundColor
+                    "yt_white1", "yt_white1_opacity95", "yt_white1_opacity98", "yt_white2", "yt_white3", "yt_white4"
+                    -> lightThemeBackgroundColor
 
                     else -> continue
                 }
             }
         }
 
-        // copies the resource file to change the splash screen color
-        context.copyResources("theme",
-            ResourceUtils.ResourceGroup("values-night-v31", "styles.xml")
+        // edit the resource files to change the splash screen color
+        val paths: List<String> = listOf(
+            "res/values/attrs.xml", // resource attributes list for the app
+            "res/values/styles.xml", // styles attributes list for Android 11 (and below)
+            "res/values-v31/styles.xml", // styles attributes list for Android 12 (and above)
+            "res/drawable/quantum_launchscreen_youtube.xml", // Android 11 (and below) splash screen manager for Smartphones
+            "res/drawable-sw600dp/quantum_launchscreen_youtube.xml" // Android 11 (and below) splash screen manager for Tablet
         )
 
+        context.xmlEditor[paths[0]].use { editor ->
+            val file = editor.file
+
+            (file.getElementsByTagName("resources").item(0) as Element).appendChild(
+                file.createElement("attr").apply {
+                    setAttribute("format", "reference")
+                    setAttribute("name", "splashScreenColor")
+                }
+            )
+        }
+        context.xmlEditor[paths[1]].use { editor ->
+            val file = editor.file
+
+            val childNodes = (file.getElementsByTagName("resources").item(0) as Element).childNodes
+
+            for (i in 0 until childNodes.length) {
+                val node = childNodes.item(i) as? Element ?: continue
+
+                file.createElement("item").apply {
+                    setAttribute("name", "splashScreenColor")
+
+                    appendChild(
+                        file.createTextNode(
+                            when (node.getAttribute("name")) {
+                                "Base.Theme.YouTube.Launcher.Dark" -> darkThemeBackgroundColor
+                                "Base.Theme.YouTube.Launcher.Light" -> lightThemeBackgroundColor
+                                else -> {"null"}
+                            }
+                        )
+                    )
+
+                    if (this.textContent != "null")
+                        node.appendChild(this)
+                }
+            }
+        }
```

Comment:
A lot here is duplicated with the code below. Please reduce it.

Reply:
What do you mean exactly with "duplicated"? You're referring to the names or other things?

Reply:
[This lines](https://github.com/revanced/revanced-patches/pull/1315/files#diff-5a51aa8d702357ed352218e03738bda901cd9181c815090f201b6060b753dddeR63-R68) are duplicate with [these](https://github.com/revanced/revanced-patches/pull/1315/files#diff-5a51aa8d702357ed352218e03738bda901cd9181c815090f201b6060b753dddeR89-R94) as an example.

Reply:
Should be good now. I squashed the cloned code into a single one.